### ISSUE TITLE
Rule prefer-immutable-types doesn't work with fp-ts Types

### ISSUE THREAD

**Issue by ldrick** [eslint-functional/eslint-plugin-functional#525]
## Bug Report

If you use fp-ts ReadonlyNonEmptyArray (https://gcanti.github.io/fp-ts/modules/ReadonlyNonEmptyArray.ts.html) like this:
```typescript
import {readonlyNonEmptyArray as RNEA} from 'fp-ts';

const myMethod= (
	values: RNEA.ReadonlyNonEmptyArray<number>,
): RNEA.ReadonlyNonEmptyArray<string> => RNEA.map(/* some code*/)(values)
```

### Expected behavior

The parameter `values: RNEA.ReadonlyNonEmptyArray<number>` should be detected as being immutable.

### Actual behavior

The parameter `values: RNEA.ReadonlyNonEmptyArray<number>` gets marked as being mutable.
Rewriting the above to somehow fixes the error, but looks not quite right:
```typescript
import {readonlyNonEmptyArray as RNEA} from 'fp-ts';

const myMethod= (
	values: Readonly<RNEA.ReadonlyNonEmptyArray<number>>,
): RNEA.ReadonlyNonEmptyArray<string> => RNEA.map(/* some code*/)(values)
```

### Steps to reproduce

use above sample

### Proposed changes

Maybe, as I don't know the details, somehow check, if the Type is a superior Type of immutable types?

**Comment by RebeccaStevens** [comment]
Looks like this is a bug in the upstream library. I've open an issue for it over there ([is-immutable-type#40](https://github.com/RebeccaStevens/is-immutable-type/issues/40)).

In the meantime, one possible work around is to use an [immutability override](https://github.com/eslint-functional/eslint-plugin-functional/blob/main/docs/rules/settings/immutability.md) to mark `ReadonlyNonEmptyArray` as `ReadonlyShallow`:

```json
"settings": {
  "immutability": {
    "overrides": [
      {
        "name": "ReadonlyNonEmptyArray",
        "to": "ReadonlyShallow",
        "from": "Mutable"
      }
    ]
  }
}
```

**Comment by RebeccaStevens** [comment]
Fixed in the upstream now.
